### PR TITLE
[Workspace] Refactor collaborators panel at workspace create

### PR DESCRIPTION
**Pull request by Kapian1234**
### Description
Refactor collaborators panel at workspace create

## Screenshot
before
Screenshot 2024-09-30 at 12 59 23

after
Screenshot 2024-09-30 at 16 16 13

## Changelog

- feat: Refactor collaborators panel at workspace create

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx [hunk 1]
```diff
@@ -138,28 +148,47 @@ export const WorkspaceCreatorForm = (props: WorkspaceCreatorFormProps) => {
...
-                errors={formErrors.permissionSettings?.fields}
-                onChange={setPermissionSettings}
-                permissionSettings={formData.permissionSettings}
-                disabledUserOrGroupInputIds={disabledUserOrGroupInputIdsRef.current}
-                data-test-subj={`workspaceForm-permissionSettingPanel`}
-              />
-            </>
+            <EuiSplitPanel.Outer direction="column">
+              <EuiSplitPanel.Inner>
+                <EuiTitle
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:153]: Can we use `EuiTitle + h2` as panel title [like this](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/workspace/public/components/workspace_creator/creator_details_panel.tsx#L42-L48) to align with the guidance.

Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:153]: ok

File: src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx [hunk 2]
```diff
@@ -138,28 +147,47 @@ export const WorkspaceCreatorForm = (props: WorkspaceCreatorFormProps) => {
...
-              <EuiSpacer size="m" />
-              <WorkspacePermissionSettingPanel
-                errors={formErrors.permissionSettings?.fields}
-                onChange={setPermissionSettings}
-                permissionSettings={formData.permissionSettings}
-                disabledUserOrGroupInputIds={disabledUserOrGroupInputIdsRef.current}
-                data-test-subj={`workspaceForm-permissionSettingPanel`}
-              />
-            </>
+            <EuiSplitPanel.Outer direction="column">
```
Comment (wanglam) [inline src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:150]: Any reason to use SplitPanel? Can we use `EuiPanel` instead?

Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:150]: No particular reason, I just happened to see that SplitPanel can achieve this effect. I'll change it to EuiPanel then.

Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx:150]: It seems there will be a 32px gap between `SplitPanel.Inner`. Let use `EuiPanel` with `EuiSpacer` instead.

Comment (Kapian1234) [reply src/plugins/workspace/public/components/workspace_creator/workspace_creator_form.tsx]: ok

**Comment by ruanyl** [comment]
@Kapian1234 could you resolve the conflict please

**Review by ruanyl — DISMISSED** [review]

**Comment by ruanyl** [comment]
@Kapian1234 Could you resolve the conflicts

**Comment by wanglam** [comment]
@ruanyl The conflicts has been resolved, could you help me take a look?

File: src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_panel.tsx [hunk 3]
```diff
@@ -135,31 +136,42 @@ export const WorkspacePermissionSettingPanel = ({
...
+              </EuiFormLabel>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiButtonIcon
+                color="text"
+                iconType="trash"
+                size="xs"
+                style={{ visibility: 'hidden' }}
+              />
+            </EuiFlexItem>
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_panel.tsx:170]: I am a little bit confused on this element, it contains a hidden element. Could you please elaborate more here?

Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_panel.tsx:170]: I think this is a placeholder to align column label with inputs. I think we can change to fixed width div.

Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_panel.tsx:170]: I c, thanks. A comment here would be preferred.

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_panel.tsx (hunk 3 above)
Comment (wanglam) [reply src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_panel.tsx:170]: Sure. Will add some comments above it.

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**